Articles/SwiftUI/SwiftUI UIKit Interoop/SwiftUIInteroperability/Source Files/Common/Dependency Management/DependencyProvider.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  DependencyProvider.swift
+//  SwiftUI Interoperability
+//
+
+import UIKit
+
+// MARK: DependencyProvider
+
+/// An abstraction providing convenient reference to all major app dependencies: storage, notification handlers
+protocol DependencyProvider: AnyObject {
+
+    /// An app permanent storage.
+    var permanentStorage: LocalDataService { get }
+
+    /// An app temporary storage.
+    var temporaryStorage: AppDataCache { get }
+
+    /// An authentication service.
+    var authenticationService: AuthenticationService { get }
+
+    /// A registration service.
+    var registrationService: RegistrationService { get }
+
+    /// A presentable HUD.
+    var presentableHUD: PresentableHud { get }
+
+    /// An information alert.
+    var infoAlert: InfoAlert { get }
+
+    /// An acceptance alert.
+    var acceptanceAlert: AcceptanceAlert { get }
+}
+
+// MARK: DefaultDependencyProvider
+
+final class DefaultDependencyProvider {
+
+    let defaultLocalDataService: DefaultLocalDataService
```

Comment:
We're inside `DefaultXYZ` class and I think it's implicit that all properties are defaults too. It's realy minor comment from my side but I found reading this code quite challanging due to abuse of word `default` everywhere 😉

Reply:
Yeah, `Live` sounds much better then `Default` - I'll try to apply it whenever possible.
Unfortunately, the naming of properties as `defaultMyService` is necessary to somehow distinguish them from `DependencyProvider` protocol property definitions, eg. `var myService {get}`. I was thinking of using `_` for such properties, eg. `let _myService: LiveMyService`, but it's equally unimpressive ;)